src/fsw/FCCode/ADCSBoxMonitor.cpp

```diff
@@ -8,24 +8,52 @@ ADCSBoxMonitor::ADCSBoxMonitor(StateFieldRegistry &registry,
     unsigned int offset, Devices::ADCS &_adcs)
     : TimedControlTask<void>(registry, "adcs_monitor", offset),
     adcs_system(_adcs),
-    rwa_speed_rd_sr(adcs::rwa::min_speed_read, adcs::rwa::max_speed_read, 8*3), //full res is 16*3
+    rwa_speed_rd_sr(adcs::rwa::min_speed_read, adcs::rwa::max_speed_read, 16*3), //referenced from I2C_Interface.doc
+    rwa_speed_rd_component_sr(adcs::rwa::min_speed_read,adcs::rwa::max_speed_read, 8),
     rwa_speed_rd_f("adcs_monitor.rwa_speed_rd", rwa_speed_rd_sr),
-    rwa_torque_rd_sr(adcs::rwa::min_torque, adcs::rwa::max_torque, 8*3), //full res is 16*3
+    rwa_speed_rd_x_f("adcs_monitor.rwa_speed_rd.x",rwa_speed_rd_component_sr),
+    rwa_speed_rd_y_f("adcs_monitor.rwa_speed_rd.y",rwa_speed_rd_component_sr),
+    rwa_speed_rd_z_f("adcs_monitor.rwa_speed_rd.z",rwa_speed_rd_component_sr),
+
+    rwa_torque_rd_sr(adcs::rwa::min_torque, adcs::rwa::max_torque, 16*3), //referenced from I2C_Interface.doc
+    rwa_torque_rd_component_sr(adcs::rwa::min_torque,adcs::rwa::max_torque, 8),
     rwa_torque_rd_f("adcs_monitor.rwa_torque_rd", rwa_torque_rd_sr),
-    ssa_mode_rd(2),
+    rwa_torque_rd_x_f("adcs_monitor.rwa_torque_rd.x", rwa_torque_rd_component_sr),
+    rwa_torque_rd_y_f("adcs_monitor.rwa_torque_rd.y", rwa_torque_rd_component_sr),
+    rwa_torque_rd_z_f("adcs_monitor.rwa_torque_rd.z", rwa_torque_rd_component_sr),
+
+    ssa_mode_rd(0,2,2), //referenced from Interface.doc
     ssa_mode_f("adcs_monitor.ssa_mode", ssa_mode_rd),
-    ssa_vec_sr(0,1,1 + SerializerConstants::min_fvsz),
+    ssa_vec_sr(-1,1,16*3), //referenced from I2C_Interface.doc
     ssa_vec_f("adcs_monitor.ssa_vec", ssa_vec_sr),
-    ssa_voltage_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 12), // Full resolution to enable scientific analysis of the sun sensing algorithm
+    ssa_voltage_sr(adcs::ssa::min_voltage_rd, adcs::ssa::max_voltage_rd, 8),
     ssa_voltages_f(),
-    mag1_vec_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 8*3), //full res is 16*3
+
+    mag1_vec_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 16*3), //referenced from I2C_Interface.doc
+    mag1_vec_component_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 8),
```

Comment:
This needs to be a 16 serializer if we want full resolution:
```suggestion
    mag1_vec_component_sr(adcs::imu::min_mag1_rd_mag, adcs::imu::max_mag1_rd_mag, 16),
```